shared/lsif/api.ts

```diff
@@ -0,0 +1,91 @@
+import gql from 'tagged-template-noop'
+import { QueryGraphQLFn } from '../util/graphql'
+import { parseGitURI } from '../util/uri'
+
+/** The response envelope for all LSIF queries. */
+export interface GenericLSIFResponse<R> {
+    repository: {
+        commit: {
+            blob: {
+                lsif: R
+            }
+        }
+    }
+}
+
+/**
+ * Envelopes an LSIF resolver request.
+ *
+ * @param inner The GraphQL fragment inside of the lsif resolver.
+ */
+export const lsifRequest = (inner: string): string => gql`
+    repository(name: $repository) {
+        commit(rev: $commit) {
+            blob(path: $path) {
+                lsif {
+                    ${inner}
+                }
+            }
+        }
+    }
+`
+
+export const resourceFragment = gql`
+    resource {
```

Comment:
You could use the GraphQL fragment syntax here, which makes this string formattable by Prettier and even validatable & autocompletable (although I think that tooling is not setup  in this repo). Still makes it explicit on what type these fields apply to :)

https://graphql.org/learn/queries/#fragments

Reply:
Updated, added tooling.